### ISSUE TITLE
Trumbowyg Giphy Plugin is broken due to mixed result set from Giphy API

### ISSUE THREAD

**Issue by synth** [Alex-D/Trumbowyg#1243]
We have noticed the Giphy plugin is broken with some Giphy API responses. By broken, I mean that no images load and I see in the js console below. In the case below, you can see it immediately without doing a search, but it also repeats for some searches, but not all. It seems the Giphy API response does not always include a `downsized` entry in the json payload. 

<img width="1653" alt="Screen Shot 2021-08-13 at 3 38 52 PM" src="https://user-images.githubusercontent.com/10546/129424548-6a99255d-0414-449d-9052-c4f8f5a64ed9.png">

When testing locally, I can see the api response: 

```json
{
   "original":{
      "height":"640",
      "width":"640",
      "size":"15683861",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/giphy.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=giphy.gif&ct=g",
      "mp4_size":"2713472",
      "mp4":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/giphy.mp4?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=giphy.mp4&ct=g",
      "frames":"277",
      "hash":"86271fd2c792723e7d72b0f092c9766e"
   },
   "downsized_large":{
      "height":"318",
      "width":"318",
      "size":"5304292",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/giphy-downsized-large.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=giphy-downsized-large.gif&ct=g"
   },
   "downsized_medium":{
      "height":"279",
      "width":"279",
      "size":"3495979",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/giphy-downsized-medium.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=giphy-downsized-medium.gif&ct=g"
   },
   "fixed_height":{
      "height":"200",
      "width":"200",
      "size":"4725830",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/200.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=200.gif&ct=g"
   },
   "fixed_height_downsampled":{
      "height":"200",
      "width":"200",
      "size":"177570",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/200_d.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=200_d.gif&ct=g"
   },
   "fixed_height_still":{
      "height":"200",
      "width":"200",
      "size":"28870",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/200_s.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=200_s.gif&ct=g"
   },
   "fixed_width":{
      "height":"200",
      "width":"200",
      "size":"4725830",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/200w.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=200w.gif&ct=g",
      "webp_size":"1086370",
      "webp":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/200w.webp?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=200w.webp&ct=g"
   },
   "fixed_width_downsampled":{
      "height":"200",
      "width":"200",
      "size":"177570",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/200w_d.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=200w_d.gif&ct=g"
   },
   "fixed_width_still":{
      "height":"200",
      "width":"200",
      "size":"28870",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/200w_s.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=200w_s.gif&ct=g"
   },
   "original_still":{
      "height":"640",
      "width":"640",
      "size":"118414",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/giphy_s.gif?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=giphy_s.gif&ct=g"
   },
   "original_mp4":{
      "height":"480",
      "width":"480",
      "mp4_size":"2713472",
      "mp4":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/giphy.mp4?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=giphy.mp4&ct=g"
   },
   "480w_still":{
      "height":"480",
      "width":"480",
      "size":"15683861",
      "url":"https://media4.giphy.com/media/YRtLgsajXrz1FNJ6oy/480w_s.jpg?cid=4a2a0f59efjstu3w6te4wcmmmcqqojmg43s28v7xz14mzin9&rid=480w_s.jpg&ct=g"
   }
}
```
<img width="1358" alt="Screen Shot 2021-08-13 at 3 42 11 PM" src="https://user-images.githubusercontent.com/10546/129424711-ec168c39-425a-4675-b6ee-4d62939d5dce.png">

**Comment by synth** [comment]
Fairly easy fix: 
```
+++ b/app/assets/javascripts/3p/trumbowyg/trumbowyg.giphy.js
@@ -53,10 +53,12 @@

     var html = response.data
       .filter(function (gifData) {
-        return gifData.images.downsized.url !== '';
+        var downsized = gifData.images.downsized || gifData.images.downsized_medium;
+        return downsized.url !== '';
       })
       .map(function (gifData) {
-        var image = gifData.images.downsized,
+        var downsized = gifData.images.downsized || gifData.images.downsized_medium;
+        var image = downsized,
             imageRatio = image.height / image.width;

         return '<div class="img-container"><img src=' + image.url + ' width="' + width + '" height="' + imageRatio * width + '" loading="lazy" onload="this.classList.add(\'tbw-loaded\')"/></div>';
```

**Comment by synth** [comment]
@Alex-D any thoughts on this? We're hitting this again because 1) we're upgrading and need to carry our manual patch over and 2) it appears Giphy changed their api again so now `downsized` will always be there but may be empty so my patch needs updating. I think there should just be more robustness in general here. 🙏

**Comment by Alex-D** [comment]
I think this needs a fix that manage all possible error cases, but this issue was forgotten ^^'

**Comment by Alex-D** [comment]
Fixed in 2.26.0
